velox/exec/fuzzer/AggregationFuzzer.cpp

```diff
@@ -1057,5 +1114,56 @@ bool AggregationFuzzer::compareEquivalentPlanResults(
   }
 }
 
+bool AggregationFuzzer::verifyDistinctAggregation(
+    const std::vector<std::string>& groupingKeys,
+    const std::vector<std::string>& aggregates,
+    const std::vector<std::string>& masks,
+    const std::vector<RowVectorPtr>& input,
+    bool customVerification,
+    const std::vector<std::shared_ptr<ResultVerifier>>& customVerifiers) {
+  const auto firstPlan = PlanBuilder()
+                             .values(input)
+                             .singleAggregation(groupingKeys, aggregates, masks)
+                             .planNode();
+
+  if (customVerification) {
+    initializeVerifiers(firstPlan, customVerifiers, input, groupingKeys);
+  }
+
+  SCOPE_EXIT {
+    if (customVerification) {
+      resetCustomVerifiers(customVerifiers);
+    }
+  };
+
+  // Create all the plans upfront.
+  std::vector<PlanWithSplits> plans;
+  plans.push_back({firstPlan, {}});
+
+  // Alternate between using Values and TableScan node.
+
+  const auto inputRowType = asRowType(input[0]->type());
+  if (isTableScanSupported(inputRowType) && vectorFuzzer_.coinToss(0.5)) {
+    auto directory = exec::test::TempDirectoryPath::create();
```

Comment:
Hmm... How does this work? The directory will be deleted at the end of the 'if' block and I expect the query plan will than fail, no?